### PR TITLE
[#169185930] Add LoadingSpinnerOverlay component

### PR DESCRIPTION
**Pull request by morrys**
This PR introduces the LoadingSpinnerOverlay component.
It is used by the HOC withLoadingSpinner and the SelectLogoutOption component.

### CODE REVIEW

File: ts/components/LoadingSpinnerOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  cancelButtonStyle: {
+    alignSelf: "center"
+  }
+});
+
+type Props = Readonly<{
+  isLoading: boolean;
+  loadingCaption?: string;
+  loadingOpacity?: number;
+  onCancel?: any;
```
Comment (valentinacala) [inline ts/components/LoadingSpinnerOverlay.tsx]: what about declaring it as function?
```suggestion
  onCancel?: () => void;
```

File: ts/components/SelectLogoutOption.tsx [hunk 2]
```diff
@@ -68,42 +68,40 @@ class SelectLogoutOption extends React.PureComponent<Props, State> {
...
-          {this.props.header || null}
-          <List>
-            <ListItemComponent
-              title={I18n.t("profile.logout.cta.keepData.title")}
-              subTitle={I18n.t("profile.logout.cta.keepData.description")}
-              onPress={() => this.logout({ keepUserData: true })}
-              useExtendedSubTitle={true}
-            />
+    return (
+      <LoadingSpinnerOverlay isLoading={this.state.isLoading}>
```
Comment (valentinacala) [inline ts/components/SelectLogoutOption.tsx]: I suggest here to come back to the previous approach: with the applied changes (on this file) you stop using the HOC component and you need to explicitly wrap the content of the screen (`<Container>...</Container>`) into the `<LoadingSpinnerOverlay>`, while it is mean to be the scope of `withLoadingSpinner`

File: ts/components/LoadingSpinnerOverlay.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+type Props = Readonly<{
+  isLoading: boolean;
+  loadingCaption?: string;
+  loadingOpacity?: number;
+  onCancel?: () => void;
+}>;
+
+/**
+ * A HOC to display and overlay spinner conditionally
```
[the remark this answers is not in the document]
Comment (morrys) [reply ts/components/LoadingSpinnerOverlay.tsx]: Yes, i fixed the comment

**Comment by digitalcitizenship** [comment]
## Affected stories

  * ⚙️ [#169185930](https://www.pivotaltracker.com/story/show/169185930): Creare il componente LoadingSpinner estraendo la logica dall'HOC withLoadingSpinner

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**